Assets/MRTK/Services/InputSystem/FocusProvider.cs

```diff
@@ -1226,40 +1226,50 @@ private static void QueryScene(IMixedRealityPointer pointer, IMixedRealityRaycas
-
-                                for (int colliderIndex = 0; colliderIndex < numColliders; colliderIndex++)
+                                int numColliders = UnityEngine.Physics.OverlapSphereNonAlloc(pointer.Rays[i].Origin, pointer.SphereCastRadius, colliders, layerMask);
+                                totalNumColliders += numColliders;
+                                if (numColliders > 0)
                                 {
-                                    Collider collider = colliders[colliderIndex];
-                                    // Policy: in order for an collider to be near interactable it must have
-                                    // a NearInteractionGrabbable component on it.
-                                    // FIXME: This is assuming only the grab pointer is using SceneQueryType.SphereOverlap,
-                                    //        but there may be other pointers using the same query type which have different semantics.
-                                    if (collider.GetComponent<NearInteractionGrabbable>() == null)
+                                    if (totalNumColliders >= maxQuerySceneResults)
                                     {
-                                        continue;
+                                        Debug.LogWarning($"Maximum number of {totalNumColliders} colliders found in FocusProvider overlap query. Consider increasing the focus query buffer size in the input profile.");
                                     }
-                                    // From https://docs.unity3d.com/ScriptReference/Collider.ClosestPoint.html
-                                    // If location is in the collider the closestPoint will be inside.
-                                    Vector3 closestPointToCollider = collider.ClosestPoint(testPoint);
-                                    float distance = (testPoint - closestPointToCollider).sqrMagnitude;
-                                    if (distance < closestDistance)
+                                    for (int colliderIndex = 0; colliderIndex < numColliders; colliderIndex++)
                                     {
-                                        closestDistance = distance;
-                                        closest = collider.gameObject;
-                                        objectHitPoint = closestPointToCollider;
+                                        Collider collider = colliders[colliderIndex];
+                                        // Policy: in order for an collider to be near interactable it must have
+                                        // a NearInteractionGrabbable component on it.
+                                        // FIXME: This is assuming only the grab pointer is using SceneQueryType.SphereOverlap,
+                                        //        but there may be other pointers using the same query type which have different semantics.
```

Comment:
Might be good to do an assert of the assumption here, or print a warning message if somebody is using sphere overlap but the pointer is not a grab pointer.

Reply:
There doesn't seem to be a good way to extract that exact pointer implementation from the interface so shelving this for now.

Reply:
In the past we've done checks like `is GrabPointer`